tests/test_peak_processing.py

```diff
@@ -87,3 +86,161 @@ def test_sum_waveform(records, peak_left, peak_length):
+        r_copy = r.copy()
+        r_copy['time'] = r_copy['time'] + t_max * i * gap_factor
+        r_buffer.append(r_copy)
+        if r_copy['time'][-1] - r['time'][0] > 1.5 * magic_overflow_time:
+            # No need to go over and beyond. We should get int-overflow
+            break
+    r = np.concatenate(r_buffer)
+
+    # Do peak finding!
+    hits = strax.find_hits(r, min_amplitude=0)
+    assert len(hits)
+    hits = strax.sort_by_time(hits)
+
+    # Dummy to_pe
+    to_pe = np.ones(max(r['channel'])+1)
+
+    try:
+        # Find peaks, we might end up with negative dt here!
+        p = strax.find_peaks(hits, to_pe,
+                             gap_threshold=gap_threshold,
+                             left_extension=left_extension,
+                             right_extension=right_extension,
+                             max_duration=max_duration,
+                             # Due to these settings, we will start merging
+                             # whatever strax can get its hands on
+                             min_area=0.,
+                             min_channels=1, )
+    except AssertionError as e:
+        if not gap_threshold > left_extension + right_extension:
+            print(f'Great, we are getting the assertion statement for the '
+                  f'incongruent extensions')
+            return
+        elif not left_extension + max_duration + right_extension < magic_overflow_time:
+            # Ending up here is the ultimate goal of the tests. This
+            # means we are hitting github.com/AxFoundation/strax/issues/397
+            print(f'Great, the test worked, we are getting the assertion '
+                  f'statement for the int overflow')
+            return 
+        else:
+            # The error is caused by something else, we need to re-raise
+            raise e
+
+    if len(p) == 0:
+        print(f'rec length {len(r)}')
+    assert len(p)
+    assert np.all(p['dt'] > 0)
+
+    # Double check that this error should have been raised.
+    if not gap_threshold > left_extension + right_extension:
+        raise ValueError(f'No assertion error raised! Working with'
+                         f'{gap_threshold} {left_extension + right_extension}')
+
+    # Compute basics
+    strax.sum_waveform(p, r, to_pe)
+    strax.compute_widths(p)
+
+    try:
+        peaklets = strax.split_peaks(
+            p, r, to_pe,
+            algorithm='natural_breaks',
+            threshold=retrun_1,
+            split_low=True,
+            filter_wing_width=70,
+            min_area=0,
+            do_iterations=2)
+    except AssertionError as e:
+        if not left_extension + max_duration + right_extension < magic_overflow_time:
+            # Ending up here is the ultimate goal of the tests. This
+            # means we are hitting github.com/AxFoundation/strax/issues/397
+            print(f'Great, the test worked, we are getting the assertion '
+                  f'statement for the int overflow')
```

Comment:
Would be very strange if we ever trigger this.

Reply:
Yeah, this was the design of the tests as I first wrote the test and then implemented the solution.

Reply:
But the data stays the same is not it. It is something like result['data'][:] = r['data']